### DISCUSSION TITLE
Problem Parsing ORU

### DISCUSSION THREAD

**Discussion by tiagofaria** [nHapiNET/nHapi#486] [Q&A]
I have this ORU message

```
MSH|^~\&|APL|CMFL|APL2|CMFL|20230908152100||ORU^R01|055eec95-3ac9-4d1a-95d8-e09e8e223e8b|P|2.5|||AL
PID||1092902|1092902^^^CB^NS|^^^N_BI|ROSA^ABEL JOÃO^||19411015|M|||AV JOÃO PAULO II LT 544 3 D^^N/A^^1900-000^||^^^^^^|^^^^^^||||||||||||||||N
NTE|1|O|teste|1R
PV1||Consultas|||||||||null||||||||V.OP000002511|||S
ORC|SC|TH70|TCMFL2023000005||CM||1.000
OBR||TH70|TCMFL2023000005|26390-5^Rx Mão - direita, 2incidências|||20230908152100|||||||||||TCMFL2023000005|||CR|||CR|||1^^^|||||228549868|228549868|228549868&&medicoteste||||||||||||TCMFL2023000005^^ACCESSION_NUMBER
OBX|1|ST|^SR Instance UID||2.25.1890683.1.1||||||F||||||
```

When I parse with NonGreedyMode  = true, occurs the error 
`ORC does not exist in the group NHapi.Model.V25.Group.ORU_R01_PATIENT_RESULT`

My code:
```csharp
PipeParser parser = new PipeParser();
var msg = parser.Parse(messageStr, new ParserOptions(){NonGreedyMode = true, InvalidObx2Type = "ST" });
```

Why does the parser try to fetch the ORC segment from the Patient_Result?

**Comment by milkshakeuk** [comment]
@tiagofaria I will take a look at this but can't promise I will be able to do this at any pace.

**Comment by milkshakeuk** [comment]
@tiagofaria the same thing happens in hapi too (where nhapi was ported from).

![image](https://github.com/nHapiNET/nHapi/assets/1087376/a61805f1-3a61-4347-a7cf-d875a0d5cb7b)

**Comment by milkshakeuk** [comment]
@tiagofaria I have raised an [issue](https://github.com/hapifhir/hapi-hl7v2/issues/105) for `hapi` for this aswell.

**Comment by tiagofaria** [comment]
Thanks

On Tue, Oct 17, 2023 at 2:18 PM Jake Aitchison ***@***.***>
wrote:

> @tiagofaria <https://github.com/tiagofaria> I have raised an issue
> <https://github.com/hapifhir/hapi-hl7v2/issues/105> for hapi for this
> aswell.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nHapiNET/nHapi/discussions/486#discussioncomment-7303989>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB7SSSCT6M42M67AUKEAZ6LX72AQZAVCNFSM6AAAAAA46RB5IOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGMBTHE4DS>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Esta mensagem é ecológica se não a imprimir.
Poupe papel e ajude o ambiente.
